apps/sports-helsinki/src/domain/search/eventSearch/hooks/useSearchPage.tsx

```diff
@@ -0,0 +1,110 @@
+import type { EventTypeId } from 'events-helsinki-components';
+import {
+  useSearchTranslation,
+  useIsSmallScreen,
+  useEventListQuery,
+  getLargeEventCardId,
+} from 'events-helsinki-components';
+import { useRouter } from 'next/router';
+import React from 'react';
+import { useConfig } from 'react-helsinki-headless-cms';
+import { scroller } from 'react-scroll';
+import { toast } from 'react-toastify';
+import type { ISearchPage } from 'domain/search/combinedSearch/types';
+import { removeQueryParamsFromRouter } from 'utils/routerUtils';
+import { SEARCH_ROUTES } from '../../../../constants';
+import { getNextPage } from '../utils';
+import useEventSearchFilters from './useEventSearchFilters';
+
+function useSearchPage({ eventType }: { eventType: EventTypeId }): ISearchPage {
+  const { t } = useSearchTranslation();
+  const router = useRouter();
+  const [isFetchingMore, setIsFetchingMore] = React.useState(false);
+  const isSmallScreen = useIsSmallScreen();
+  const { meta } = useConfig();
+  const eventFilters = useEventSearchFilters(eventType);
+
+  // Query for the primary search / active search tab
+  const {
+    data: eventsData,
+    fetchMore,
+    loading: isLoadingEvents,
+  } = useEventListQuery({
+    ssr: false,
+    variables: eventFilters,
+  });
+
+  const handleLoadMore = async () => {
+    const page = eventsData?.eventList.meta
+      ? getNextPage(eventsData.eventList.meta)
+      : null;
+    setIsFetchingMore(true);
+    if (page) {
+      try {
+        await fetchMore({
+          variables: {
+            page,
+          },
+        });
+      } catch (e) {
+        toast.error(t('search:errorLoadMore'));
+      }
+    }
+    setIsFetchingMore(false);
+  };
+
+  const scrollToResultList = React.useCallback(() => {
```

Comment:
Here is great (but long) article about them https://kentcdodds.com/blog/usememo-and-usecallback
(i have to admit that I overused callbacks and memos in kesaseteli, and nowadays I tried to avoid to use them 🙄 )

Reply:
the use of useCallback was preferred by the linter! I wouldn't self have even thought about that 😊

Reply:
in the article that you shared, wasn't the point that you should not use them when there are no dependencies for the hook?

Reply:
I think the point was not to use them before there is clearly slow rerendering happening on the component/page. I think that the linter suggests to use useCallback when you have useMemo which uses a function which is not wrapped in useCallback. But then another option is to remove useMemo.

Reply:
I removed the useCallbacks from the `scrollToResultList` and `initialPageOnLoad`. From both of the search pages. I self tested it and it seemed to work OK. 👍